src/comm/ModDevPoll.cc

```diff
@@ -232,11 +226,11 @@ Comm::SetSelect(int fd, unsigned int type, PF * handler, void *client_data, time
     if (!F->flags.open) {
         /* remove from poll set */
         comm_update_fd( fd, POLLREMOVE );
-        devpoll_state[fd].state = 0;
+        F->epoll_state = 0;
         return;
     }
 
-    pollfd_events_t state_old = devpoll_state[fd].state;
+    pollfd_events_t state_old = (pollfd_events_t) F->epoll_state;
```

Comment:
In this PR?  It just avoided having to decide what the right size would be.

In another PR it helps solve https://bugs.squid-cache.org/show_bug.cgi?id=5314 in which some code paths are trying to access `devpol_state` when `Comm::SelectLoopInit()` had not yet been called, but the `fde` struct was properly initialized.  I think the epoll code would similarly crash if you just changed it to use a similar allocation strategy.

I can switch back to leaving the old allocation for now and just adjusting it to Squid_MaxFD for this PR.

Reply:
> In this PR? It just avoided having to decide what the right size would be.

Understood. Since removing that array (in this PR) creates more problems than it solves, I think you made the right call to restore it.

> In another PR...

Let's deal with that other problem in another PR. On the surface of it, I doubt that removing this array will solve that initialization problem there, but we do not need to discuss it _here_.